vms/proposervm/state_syncable_vm_test.go

```diff
@@ -564,12 +565,28 @@ func TestStateSummaryAcceptOlderBlock(t *testing.T) {
 
 	summary, err := vm.GetStateSummary(context.Background(), reqHeight)
 	require.NoError(err)
+	require.Equal(summary.Height(), reqHeight)
 
-	// test Accept skipped
+	// test Accept summary invokes innerVM
+	calledInnerAccept := false
```

Comment:
why the nested callback override? 

Why not do:

```
	innerVM.LastAcceptedF = func(context.Context) (ids.ID, error) {
		return innerSummary.ID(), nil
	}
	innerVM.GetBlockF = func(context.Context, ids.ID) (snowman.Block, error) {
		return innerBlk, nil
	}
	innerSummary.AcceptF = func(context.Context) (block.StateSyncMode, error) {
		calledInnerAccept = true
		return block.StateSyncStatic, nil
	}
```

?

Reply:
It is only after `Accept` has been invoked that the other two functions should be updated, so I override them within `AcceptF`